pkg/types/v1/file_existence.go

```diff
@@ -32,6 +32,23 @@ type FileExistenceTest struct {
 	Permissions string `yaml:"permissions"` // expected Unix permission string of the file, e.g. drwxrwxrwx
 }
 
+func (fe FileExistenceTest) MarshalYAML() (interface{}, error) {
+	return FileExistenceTest{ShouldExist: true}, nil
+}
+
+func (fe *FileExistenceTest) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type feAlias FileExistenceTest
+	feTest := feAlias{
```

Comment:
Why do we need the alias and cast below?

Reply:
If we call the unmarshal function on FileExistenceTest struct itself, it goes in to recursion as UnmarshalYAML is called again.
You want to set the default for fields first then call unmarshal function to set the values in the yaml text. Hence an Alias is necessary

Reply:
usually, this is not required if you have to validate fields. I can't avoid using Alias since you dont have access to raw text in UnmarshalYAML to check if `shouldExist` is present in the raw text.

Reply:
Ah, makes sense. Could you leave a comment explaining this?

Reply:
done